### PR TITLE
RES-1166: Fix batch norm initialization for nupic.research models

### PR DESCRIPTION
**Pull request by lscheinkman**

### CODE REVIEW

File: projects/imagenet/imagenet_experiment.py [hunk 1]
```diff
@@ -421,6 +434,10 @@ def validate(self, loader=None):
             criterion=self.loss_function,
             batches_in_epoch=self.batches_in_epoch,
         )
+        results.update(
+            learning_rate=self.get_lr()[0],
```
Comment (subutai) [inline projects/imagenet/imagenet_experiment.py]: `self.lr_scheduler.get_lr()[0]`

??

File: projects/imagenet/imagenet_experiment.py [hunk 2]
```diff
@@ -172,51 +176,69 @@ def _create_optimizer(model, optimizer_class, optimizer_args, optimizer_groups=N
...
+            m.bn3.weight = nn.Parameter(torch.zeros_like(m.bn3.weight))
+        elif isinstance(m, (
+            nupic.research.frameworks.pytorch.models.resnets.BasicBlock,
+            nupic.research.frameworks.pytorch.models.resnets.Bottleneck
+        )):
+            # initialized the last BatchNorm in each BasicBlock to 0
+            *_, last_bn = filter(lambda x: isinstance(x, nn.BatchNorm2d),
+                                 m.regular_path)
+            last_bn.weight = nn.Parameter(torch.zeros_like(last_bn.weight))
+        elif isinstance(m, nn.Linear):
```
Comment (subutai) [inline projects/imagenet/imagenet_experiment.py]: This is not batch norm. It's initializing linear weights.

File: projects/imagenet/imagenet_experiment.py (hunk 1 above)
Comment (lscheinkman) [reply projects/imagenet/imagenet_experiment.py]: `self.get_lr()` gets the actual LR used by the optimizer

Comment (subutai) [reply projects/imagenet/imagenet_experiment.py]: Ah ok.

File: projects/imagenet/imagenet_experiment.py (hunk 2 above)
Comment (lscheinkman) [reply projects/imagenet/imagenet_experiment.py]: That is correct, maybe the name is misleading, but in the paper describing the BN initialization they initialize the linear layers using a gaussian distribution with zero mean and 0.01 standard
deviation. They claim that helped with large minibatches.

Comment (subutai) [reply projects/imagenet/imagenet_experiment.py]: 👍
